src/bin/miri.rs

```diff
@@ -554,8 +554,22 @@ fn main() {
         } else if arg == "-Zmiri-disable-stacked-borrows" {
             miri_config.borrow_tracker = None;
         } else if arg == "-Zmiri-tree-borrows" {
-            miri_config.borrow_tracker = Some(BorrowTrackerMethod::TreeBorrows);
+            miri_config.borrow_tracker =
+                Some(BorrowTrackerMethod::TreeBorrows(TreeBorrowsParams {
+                    precise_interior_mut: true,
+                }));
             miri_config.provenance_mode = ProvenanceMode::Strict;
+        } else if arg == "-Zmiri-tree-borrows-no-precise-interior-mut" {
+            if matches!(miri_config.borrow_tracker, Some(BorrowTrackerMethod::TreeBorrows { .. })) {
+                miri_config.borrow_tracker =
+                    Some(BorrowTrackerMethod::TreeBorrows(TreeBorrowsParams {
+                        precise_interior_mut: false,
+                    }));
```

Comment:
The more idiomatic way to do this is to capture a reference to the options inside `miri_config.borrow_tracker` and mutate them in-place. This also makes the logic more robust as it will then remain correct when more options are added in the future.